Apps/Inc/Tasks.h

```diff
@@ -46,12 +44,14 @@
 #define TASK_DEBUG_DUMP_STACK_SIZE          DEFAULT_STACK_SIZE
 #define TASK_COMMAND_LINE_STACK_SIZE        DEFAULT_STACK_SIZE
 
+/**
+ * Task error variable type
+*/
+typedef int16_t error_code_t;
```

Comment:
Should this be signed or unsigned?

Reply:
I chose signed because the error enum for Update Display uses signed numbers, but if we change or remove the enum, then we can change this to unsigned.

Reply:
Ah I see, since our errors are negative there. Could we then change the other error enums to use negative numbers?

Reply:
I suppose we could, but they currently all have various reasons for being different:

- ReadCarCAN.c uses the default enum values as a stylistic choice so that the enum can be changed without having to assign or reassign values.
- UpdateDisplay uses negative numbers because of (I think) convention.
- ReadTritium actually uses uint16_t values that are numbered by bit-shifting left. I think this is because that's how the tritium does errors (though I'm not sure).

Another thing to note is that Display_Error currently takes in uint8_t values, so we'd have to revise the function if we want to use longer values.

If we standardize them, which system would be your preference?

Reply:
Let's do default enum values for all of them, it's much easier.

Reply:
On second thought, changing ReadTritium's errors to use the default enum would require adding something to convert the Tritium's errors to the default enum values. Currently, the motor errors are stored straight into the bitmap and then asserted directly like this: 
`Motor_FaultBitmap = *((uint16_t*)(&dataBuf.data[4])); //Storing error flags into Motor_FaultBitmap
					assertTritiumError(Motor_FaultBitmap);`

I could add a switch case or write a function to convert bit-shifted values to the numbered values, but would that be worth it or just unnecessarily convoluted?

Reply:
Ohh I didn't realize it was acting as a bitmap. Currently we only assert one error, even if there are multiple in the bitmap, but it works out alright because every error except hall sensor is nonrecoverable and gets handled the same way. However, I'll make a note so that if we add more recoverable errors, we'll know we need to make additional checks to catch them all.